pkg/execution/svccatadmission/internaldns.go

```diff
@@ -78,6 +78,20 @@ func InternalDNSAdmitFunc(ctx context.Context, microsServerClient microsServerCl
 		return nil, errors.Errorf("error fetching service central data for serviceName %q", serviceName)
 	}
 
+	if serviceCentralData == nil {
+		reason := fmt.Sprintf(
+			"namespace service %q does not exist in Service Central - should be impossible",
+			serviceName)
+		return &admissionv1beta1.AdmissionResponse{
+			Allowed: false,
+			Result: &metav1.Status{
+				Message: reason,
+				Code:    http.StatusForbidden,
```

Comment:
Worth double checking: we had an issue recently when webhook would reject request which deleted finalizer from the object (i.e. GC was proceeding with deletion). If we ran into the state of missing Service Central record, we should make sure that we don't block the deletion?

Reply:
Also, we could probably return 422 code instead here?

Reply:
GC is covered by shared registration code for the webhooks. 422 sounds fine.